Consequences/Result.cs

```diff
@@ -0,0 +1,18 @@
+﻿namespace Consequences;
+public struct Result<T> 
+{
+  private ResultItem<T>[] results;
```

Comment:
results should probably be a readonly property Results. That way we can get the array of results out of the result object and add to it.